### PR TITLE
Fix price parsing of prices with "nbsp" characters

### PR DESCRIPTION
**Pull request by vegaro**
We had a test that covered this, but the code behaves differently in an Android device and during unit testing.

I thought I was crazy, but https://stackoverflow.com/questions/27402616/how-to-get-plain-price-value-with-amazon-in-app-purchasing-api#comment43795089_27449971 this comment corroborates my theory.

We have to migrate this test to be instrumented but I encountered a bunch of issues doing that:

- Adding a simple instrumented test brings the number of methods above 64k, which means we need to enable multidex.
- After enabling multidex I started getting Java out of memory errors when compiling.
- I gave more memory allocation to the JVM and then Kotlin started to fail so I had to increase the Kotlin version
- I looked into some dependencies we have using a dependencies plugin and realized that we have a bunch of dependencies that are not used, specially  testing dependencies. I managed to remove some but the +64k methods were still there
- So then when I ran the project with multidex, updated dependencies and more memory, I noticed how executing connectedAndroidTest gradle task also runs the integration tests, because they are part of the project and connectedAndroidTest runs all tests in the project.
- I thought about moving these new tests into the integration-test app but the function I am testing is not public, plus this is not really an integratino test where we hit our backend.

In summary, the solution I think it is to extract the integration-test and the purchase-tester apps outside of the project and into their own projects. This will probably reduce the number of methods by a lot, compilation time too and will let us ran the instrumented tests of the project without running the integration tests too.

This will take me some time to adjust because of CI, and having a relative dependency to purchases-android (which I think we can do with includeBuild but I am not sure).

### CODE REVIEW

File: feature/amazon/src/main/java/com/revenuecat/purchases/amazon/storeProductConversions.kt [hunk 1]
```diff
@@ -78,7 +78,11 @@ internal fun String.parsePriceUsingRegex(): BigDecimal? {
     val matcher = pattern.matcher(this)
     return matcher.takeIf { it.find() }?.let {
         val dirtyPrice = matcher.group()
-        var price = dirtyPrice.replace(" ", "")
+        var price =
+            dirtyPrice.replace(" ", "")
+            .replace(" ", "")
+            .replace("${Typography.nbsp}", "")
+        price = price.trim()
```
Comment (aboedo) [inline feature/amazon/src/main/java/com/revenuecat/purchases/amazon/storeProductConversions.kt:85]: let's add a comment here explaining why this is needed and pointing to the SO post

**Review by aboedo — APPROVED** [review]
Looks great! But let's add a comment so the next dev maintaining this knows not to remove it. 
Good to go after that! 
Great job figuring this one out 💪

File: feature/amazon/src/main/java/com/revenuecat/purchases/amazon/storeProductConversions.kt [hunk 2]
```diff
@@ -78,7 +78,11 @@ internal fun String.parsePriceUsingRegex(): BigDecimal? {
     val matcher = pattern.matcher(this)
     return matcher.takeIf { it.find() }?.let {
         val dirtyPrice = matcher.group()
-        var price = dirtyPrice.replace(" ", "")
+        var price =
+            dirtyPrice.replace(" ", "")
+            .replace(" ", "")
```
Comment (NachoSoto) [inline feature/amazon/src/main/java/com/revenuecat/purchases/amazon/storeProductConversions.kt:83]: Looks like one of these is a tab? Maybe document that so it doesn't seem like it's duplicated 😅
